search/search-domain/src/main/java/io/camunda/search/filter/TenantFilter.java

```diff
@@ -7,11 +7,18 @@
  */
 package io.camunda.search.filter;
 
+import io.camunda.search.entities.TenantMemberEntity.TenantMemberType;
 import io.camunda.util.ObjectBuilder;
 import java.util.Set;
 import java.util.function.Function;
 
-public record TenantFilter(Long key, String tenantId, String name, Set<Long> memberKeys)
+public record TenantFilter(
+    Long key,
+    String tenantId,
+    String name,
+    String joinParentId,
+    TenantMemberType tenantMemberType,
+    Set<Long> memberKeys)
```

Comment:
```suggestion
    String joinParentId,
    TenantMemberType tenantMemberType,
    Set<Long> memberKeys)
```
We could extract those into a `TenantMemberFilter`, which is nicer because it separates the two use-cases of the tenant filter. On the other hand, it's another class and the type union pattern is used in many places already.

Reply:
Oh, nevermind, they can actually be used as a combined filter.